### DISCUSSION TITLE
Did you try to connect from UART or USB driver

### DISCUSSION THREAD

**Discussion by caglarsayin** [bitrate16/ppa6-python#10] [General]
Hello since we know the chip at32f413, did you try to connect to the terminal? I think we can export the relevant code from there which is possibly Python.

P.S. I did not because there is no official driver from artery for Mac or Linux

**Comment by bitrate16** [comment]
I still haven't reversed the USB based protocols, but if I'll have a motivation to, I can do it as based on decompiling/tracing stock driver

**Comment by caglarsayin** [comment]
I tried to connect through UART and seems like it is off

**Reply by bitrate16** [reply]
You can try to sniff USB transmission on stock driver via wireshark & usbcap

**Reply by caglarsayin** [reply]
I know but I do not use Windows. It would be amazing if we can extract the firmware
